hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -1017,12 +1017,19 @@ static private void ensurePositiveInt(int val, String key) {
    *
    * @param newVal - Must be a positive non-zero integer.
    */
-  public void setMaxReplicationStreams(int newVal) {
-    ensurePositiveInt(newVal,
-        DFSConfigKeys.DFS_NAMENODE_REPLICATION_MAX_STREAMS_KEY);
+  @VisibleForTesting
+  public void setMaxReplicationStreams(int newVal, boolean ensurePositiveInt) {
+    if (ensurePositiveInt) {
```

Comment:
Why do we need to add this judgment?

Reply:
Thanks @tomscut help me review it.

Add this judgment is for compatibility with the case set maxReplicationStreams is 0 in TestBlockManager.java#testFavorDecomUntilHardLimit